Update Astro component docs (#2404)
withastro/docs: src/pages/en/core-concepts/astro-components.mdx

Astro component syntax is a superset of HTML. The syntax was [designed to feel familiar to anyone with experience writing HTML or JSX](#differences-between-astro-and-jsx), and adds support for including components and JavaScript expressions. You can spot an Astro component by its file extension: `.astro`.

> Astro components are extremely flexible. Often, an Astro component will contain some **reusable UI on the page**, like a header or a profile card. At other times, an Astro component may contain a smaller snippet of HTML, like a collection of common `<meta>` tags that make SEO easy to work with. It can even contain an entire page layout.

Reviewer: ```suggestion
Astro components are extremely flexible. Often, an Astro component will contain some **reusable UI on the page**, like a header or a profile card. At other times, an Astro component may contain a smaller snippet of HTML, like a collection of common `<meta>` tags that make SEO easy to work with. Astro components can even contain an entire page layout.
```

> The most important thing to know about Astro components is that they **render to HTML**, be it ahead of time in static builds or on-demand in [server-side rendering (SSR)](/en/guides/server-side-rendering/). Even if you run JavaScript code inside of your components, all of it will be stripped from the final page sent to your users' browsers. The result is a faster site, with zero JavaScript footprint added by default.

Reviewer 2: ```suggestion
The most important thing to know about Astro components is that they **render to HTML**, be it ahead of time in static builds or on-demand in [server-side rendering (SSR)](/en/guides/server-side-rendering/). You can include JavaScript code inside of your component frontmatter, and all of it will be stripped from the final page sent to your users' browsers. The result is a faster site, with zero JavaScript footprint added by default.
```

It isn't true that all of the JavaScript code inside of your components will be stripped; `<script>` tags are sent!

> The most important thing to know about Astro components is that they **render to HTML**, be it ahead of time in static builds or on-demand in [server-side rendering (SSR)](/en/guides/server-side-rendering/). Even if you run JavaScript code inside of your components, all of it will be stripped from the final page sent to your users' browsers. The result is a faster site, with zero JavaScript footprint added by default.

Reviewer 2: I would rephrase this paragraph to focus on **not rendering on the client**, rather than **rendering to HTML**. React components also render to HTML. Something like:

```suggestion
The most important thing to know about Astro components is that they **don't render on the client**. They render either at build-time or on-demand using [server-side rendering (SSR)](/en/guides/server-side-rendering/). You can include JavaScript code inside of your component frontmatter, and all of it will be stripped from the final page sent to your users' browsers. The result is a faster site, with zero JavaScript footprint added by default.
```

> The most important thing to know about Astro components is that they **render to HTML ahead of time**. It could be during the build step or on the server on demand, depending on how you have configured your project. Even if you run JavaScript code inside of your components, all of it will be stripped from the final page sent to your users' browsers. The result is a faster site, with zero JavaScript footprint added by default. 

Reviewer 3: Not sure about the wording here, saying "Astro components render to HTML ahead of time [....] on demand in the server" feels kinda contradictory to me? How about changing this a bit?

```suggestion
The most important thing to know about Astro components is that they **render to HTML**, be it ahead of time in static builds or on-demand in SSR. Even if you run JavaScript code inside of your components, all of it will be stripped from the final page sent to your users' browsers. The result is a faster site, with zero JavaScript footprint added by default.
```

Author: That sounds better 👍 

I guess we're allowed to use the word "SSR" here?

Reviewer 3: Great question! I feel like SSR is such a common buzzword nowadays that most people will understand we are talking about server-side rendering, although I can see the argument for using the full term instead, I'll let @sarah11918 weight on this.

Reviewer: We're allowed to say it! Often if it's at the very top of a page like this, we might spell it out in full and then also put (SSR) in brackets after. And, I would probably choose to link to the SSR page from it.

Reviewer 3: Agree!

```suggestion
The most important thing to know about Astro components is that they **render to HTML**, be it ahead of time in static builds or on-demand in [server-side rendering (SSR)](/en/guides/server-side-rendering/). Even if you run JavaScript code inside of your components, all of it will be stripped from the final page sent to your users' browsers. The result is a faster site, with zero JavaScript footprint added by default.
```

Author: Sounds good!

<p class:list={["add", "dynamic", {classNames: true}]} />
```

You're not limited to just Astro components though. You can also include **JavaScript framework components** in your template.

📚 Learn about using [JavaScript framework components](/en/core-concepts/framework-components/) in your Astro project.

## JSX-like Expressions

You can define local JavaScript variables inside of the frontmatter component script within an Astro component. You can then inject these variables into the component's HTML template using JSX-like expressions!

:::note[dynamic vs reactive]

> Using this approach, you can include **dynamic** values that are calculated in the frontmatter. But once included, these values are not **reactive** and will never change. Astro components are templates that only run once, at build time.

Reviewer: ```suggestion
Using this approach, you can include **dynamic** values that are calculated in the frontmatter.  But once included, these values are not **reactive** and will never change. Astro components are templates that only run once, at build time.
```

> Using this approach, you can include **dynamic** values that are calculated in the frontmatter. But once included, these values are not **reactive** and will never change. Astro components are templates that only run during the rendering step.

Reviewer: ```suggestion
Using this approach, you can include **dynamic** values that are calculated in the frontmatter. But once included, these values are not **reactive** and will never change. Astro components are templates that only run once, at build time.
```
Reverting this change, since in the new PR, this part of the text is in a new file. I've copied the suggestion to include in the new Astro Syntax page. This way, all edits in this PR only discuss content staying on the components page.
